task/web-apps-build-tree

edsanville opened the pull request:
Limited to only commits directly related to building all artifacts into the build/<arch>/intermediate directory.

There SHOULD be nothing generated in the src tree.

edsanville committed:
Fix install_dependencies.sh and build_messages.sh to target a specified directory

edsanville committed:
TypeScript outputs into intermediate build directory

edsanville committed:
The run.sh script serves web apps from the BUILD_DIR

edsanville committed:
run.sh symlinks jed to the build dir, so we can hot reload while developing

edsanville committed:
Add build_venv.sh script

edsanville committed:
Build jdv into a target directory

edsanville committed:
Use build directory targeting scripts for web CMake build

edsanville committed:
Copy web src trees into build/<arch>/intermediate/web, and then install node_modules there, and build apps into their final locations
Build process doesn't generate any node_modules directory anywhere in the src tree anymore

edsanville committed:
Build JCC directly into a target directory

edsanville committed:
install_dependencies.sh should use its command-line parameter
npm should be quiet unless there's an error

edsanville committed:
Fix jcc src path in run.sh

edsanville committed:
Removed old Makefile for jdv

michael-jaia commented on src/python/build_venv.sh:
```diff
@@ -0,0 +1,23 @@
+echo 🟢 Building the python venv into ${TARGET_DIR}
+
+## Make target python directory
+mkdir -p ${TARGET_DIR}
```
Do we want the -p switch here?

edsanville replied:
It needs to create all parent directories to the path, in case they don't exist, but...  now that I think of it, it might be a problem if the venv already exists and we want to clobber it, (removing all the installed packages).

edsanville replied:
If we want to clobber any pre-existing venv directories, that would be an additional "rm -rf ${TARGET_DIR}" just before the mkdir.

michael-jaia commented on src/python/build_venv.sh:
```diff
@@ -0,0 +1,23 @@
+TARGET_DIR=$1
+JAIA_DIR="$(pwd)/../../"
+
+# Build the venv
```
Should the # be the same length (I'm not sure if there is a reason for #, ##, ###)

michael-jaia replied:
Should we also make them start with a capital or lowercase letter?

edsanville replied:
I was going for a sort of "nested list" thing, like in markdown files.  I wonder if there's a better way to do it.  We should stick with capitalized though, I think.

edsanville replied:
Trying indentation for script structure, instead of multiple hashes...

michael-jaia commented on src/web/CMakeLists.txt:
```diff
@@ -1,40 +1,64 @@
   )
-add_dependencies(command_control npm_depdendencies)
+
+# Copy the JCC server
```
Make jcc lowercase to be like other instances

michael-jaia commented on src/web/command_control/webpack.config.js:
```diff
@@ -1,129 +1,113 @@
+module.exports = (env, argv) => {
+  webpackOptions =
+      [
+        // Client
```
Do we need these comments?

edsanville replied:
Nope!

michael-jaia commented on src/web/install_dependencies.sh:
```diff
@@ -1,45 +1,11 @@
-npm install --no-audit --quiet
+pushd ${PACKAGE_JSON_DIR} > /dev/null
+    # Shut up unless there's an error!
+    npm install --no-audit --no-progress --silent || npm install --no-audit --no-progress --quiet
```
Will this hide warnings about packages that have been depreciated?

edsanville replied:
Oooh, interesting point.  I tailored this script for the build.sh process, so I wanted it to be silent unless errors were present.  I eventually moved this build logic into CMakeLists.txt.

Perhaps we should catch deprecated packages when we're building for the web, using the run.sh scripts?

michael-jaia commented on src/web/jdv/client/test.webpack.config.js:
Now that we removed the "scripts" property from the package.json...can we remove this file too?

michael-jaia commented on src/web/run.sh:
```diff
@@ -5,22 +5,32 @@ set -e
+## Install pre-requisites
+./install_dependencies.sh ./
+
+## copy the webpack.config.js file to the intermediate build directory, so webpack can import from node_modules
```
Capitalize c in copy

michael-jaia commented on src/web/server/app.py:
```diff
@@ -44,12 +46,12 @@ def parseDate(date):
-root = '../command_control/dist/client/'
-jed = '../jed/'
+jcc: str = os.path.join(args.appRoot, 'jcc')
+jed = os.path.join(args.appRoot, 'jed')
```
Should : str also be added after jed for consistency?

edsanville committed:
Use indentation for script structure

edsanville committed:
Use lower-case jcc

edsanville committed:
Remove old comments

edsanville committed:
Remove unused webpack config file

edsanville committed:
Capitalize and use indentation to structure run.sh

edsanville committed:
Added str type hint for consistency

edsanville committed:
build_venv.sh should abort if anything fails
Use DCCL_DIR and GOBY_DIR, in case our .proto files aren't in the standard locations

edsanville committed:
Merge remote-tracking branch 'origin/1.y' into task/web-apps-build-tree

edsanville committed:
Merge remote-tracking branch 'origin/1.y' into task/web-apps-build-tree

edsanville committed:
Put proto_include in the /tmp directory, and remove it afterward, to avoid confusion with an additional intermediate directory

edsanville committed:
Put all of the typescript definition files into the single intermediate directory for clarity

ferrom-jaia committed:
Merge branch '1.y' into task/web-apps-build-tree

edsanville committed:
Add rsync and python3-venv as apt build dependencies in setup-tools-build-nodocker.sh

edsanville committed:
Added a test-setup-web-run.sh script

ferrom-jaia committed:
Fix merge conflicts

ferrom-jaia committed:
Merge branch '1.y' into task/web-apps-build-tree

ferrom-jaia reviewed (dismissed):
Looks great to me! Tested on my machine. 
1. Removed build directory
2. ran build.sh
3. ran the run.sh in the web dir
4. ran the all.launch

ferrom-jaia committed:
typscript folder added to gitignore and vscode settings

ferrom-jaia committed:
Merge branch 'task/web-apps-build-tree' of github.com:jaiarobotics/jaiabot into task/web-apps-build-tree

michael-jaia committed:
Use npx for webpack commands + update gitignore

michael-jaia committed:
Fix merge conflicts

michael-jaia committed:
Add one more npx for webpack command

michael-jaia commented:
Built in new Ubuntu 22.04 VM and my current WSL system without errors. Launched JCC and JDV to see runtime environment.
